app/controllers/api/services_controller.rb

```diff
@@ -25,6 +25,57 @@ def reconfigure_resource(type, id = nil, data = nil)
       end
     end
 
+    def start_resource(type, id = nil, _data = nil)
+      raise BadRequestError, "Must specify an id for starting a #{type} resource" unless id
+
+      api_action(type, id) do |klass|
+        service = resource_search(id, type, klass)
+        api_log_info("Starting #{service_ident(service)}")
+
+        begin
+          description = "#{service_ident(service)} starting"
+          task_id = queue_object_action(service, description, :method_name => "start", :role => "service_stuff")
+          action_result(true, description, :task_id => task_id)
+        rescue => e
+          action_result(false, e.to_s)
+        end
+      end
+    end
+
+    def stop_resource(type, id = nil, _data = nil)
+      raise BadRequestError, "Must specify an id for starting a #{type} resource" unless id
+
+      api_action(type, id) do |klass|
+        service = resource_search(id, type, klass)
+        api_log_info("Stopping #{service_ident(service)}")
+
+        begin
+          description = "#{service_ident(service)} stopping"
+          task_id = queue_object_action(service, description, :method_name => "stop", :role => "service_stuff")
+          action_result(true, description, :task_id => task_id)
+        rescue => e
+          action_result(false, e.to_s)
+        end
+      end
+    end
+
+    def pause_resource(type, id = nil, _data = nil)
+      raise BadRequestError, "Must specify an id for starting a #{type} resource" unless id
+
+      api_action(type, id) do |klass|
+        service = resource_search(id, type, klass)
+        api_log_info("Pausing #{service_ident(service)}")
+
+        begin
+          description = "#{service_ident(service)} pausing"
+          task_id = queue_object_action(service, description, :method_name => "suspend", :role => "service_stuff")
```

Comment:
@abellotti I noticed I sent the wrong message to `queue_object_action` and it still passed the test....is there a way to get it to fail?